dashboard/models.py

```diff
@@ -0,0 +1,166 @@
+        db_table = 'assignment_weight_consideration'
+
+class Course(models.Model):
+    id = models.CharField(primary_key=True, max_length=255, verbose_name="Canvas Course Id")
+    term_id = models.ForeignKey(AcademicTerms, verbose_name="Term Id", on_delete=models.SET_NULL, db_column='term_id')
+    name = models.CharField(max_length=255, verbose_name="Name")
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        managed = False
+        db_table = 'course'
+        verbose_name = "Course"
+
+
+class CourseViewOption(models.Model):
+    course = models.OneToOneField(Course, on_delete=models.CASCADE, primary_key=True, verbose_name="Course View Option Id")
+    show_files_accessed = models.BooleanField(blank=False, null=False, verbose_name="Show Files Accessed View")
+    show_assignment_planning = models.BooleanField(blank=False, null=False, verbose_name="Show Assignment Planning View")
+    show_grade_distribution = models.BooleanField(blank=False, null=False, verbose_name="Show Grade Distribution View")
+
+    def __str__(self):
+        return f"Course options for {self.course}"
+
+    class Meta:
+        managed = False
+        db_table = 'course_view_option'
+
+class File(models.Model):
+    id = models.CharField(primary_key=True, max_length=255, verbose_name="File Id")
+    name = models.TextField(verbose_name="File Name")
+    course_id = models.CharField(max_length=255, verbose_name="Course Id")
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        managed = False
+        db_table = 'file'
+
+
+class Submission(models.Model):
+    id = models.CharField(primary_key=True, max_length=255, verbose_name="Submission Id")
+    assignment_id = models.CharField(max_length=255, verbose_name="Assignment Id")
+    course_id = models.CharField(max_length=255, verbose_name="Course Id")
+    user_id = models.CharField(max_length=255, verbose_name="User Id")
+    score = models.CharField(max_length=255, blank=True, null=True, verbose_name="Score")
+    graded_date = models.DateTimeField(blank=True, null=True, verbose_name="Graded DateTime")
+
+    def __str__(self):
+        return "TODO"
+
+    class Meta:
+        managed = False
+        db_table = 'submission'
+
+
+class UnizinMetadata(models.Model):
+    pkey = models.CharField(primary_key=True, max_length=20, verbose_name="Key")
+    pvalue = models.CharField(max_length=100, blank=True, null=True, verbose_name="Value")
+
+    class Meta:
+        managed = False
+        db_table = 'unizin_metadata'
+
+
+class User(models.Model):
+    id = models.CharField(primary_key=True, max_length=255, verbose_name="User Id")
+    name = models.CharField(max_length=255, verbose_name="Name")
+    sis_id = models.CharField(max_length=255, blank=True, null=True, verbose_name="SIS Id")
+    sis_name = models.CharField(max_length=255, blank=True, null=True, verbose_name="SIS Name")
+    course_id = models.CharField(max_length=255, verbose_name="Course Id")
```

Comment:
course_id should also be the primary key for the cases the student is part of the 2 courses, it won't throw the id already error

Reply:
Thanks I'll double check those are correct in the model.

I only verified a few of the tables are actually correct as I'm not using everything here yet.

Reply:
In the django model this is represented by the Meta 
        unique_together = (('id', 'course_id'),)

This was already set.